libopenfpga/libfpgaaesencryption/test/AES_main.cpp

```diff
@@ -0,0 +1,211 @@
+#include "AESfunctions.h"
+#include <fstream> 
+#include <string>
+#include <sstream>
+#include <algorithm>
+#include<cctype>
+
+//String conversion function
+string convertToString(char* a, int size)
+{
+    int i;
+    string s = "";
+    for (i = 0; i < size; i++) {
+        s = s + a[i];
+    }
+    return s;
+}
+
+//Function to convert binary to Decimal
+
+int binaryToDecimal(int n)
+{
+    int num = n;
+	int dec_value = 0;
+
+    // Initializing base value to 1, i.e 2^0
+     int base = 1;
+
+    int temp = num;
+    while (temp) {
+        int last_digit = temp % 10;
+        temp = temp / 10;
+
+        dec_value += last_digit * base;
+
+        base = base * 2;
+    }
+
+    return dec_value;
+}
+
+int main()
+{  
+    // These variables allow to change the AES length
+    // Nk = 4, 6 or 8 for AES 128, 192 or 256 respectively
+    // Only set Nk and the rest will compute according to the AES specification
+    unsigned short Nk = 4; // 4 or 6 or 8 [32-bit words] columns in cipher key
+    unsigned short CipherKeyLenghth = Nk * rows; // bytes
+    unsigned short Nr = (Nk > Nb) ? Nk + 6 : Nb + 6; // = 10, 12 or 14 rounds
+    unsigned short ExtdCipherKeyLenghth = (Nr + 1) * stt_lng; // bytes in the expanded cipher key
+	/*
+	cout << "AES with Nb = " << Nb << " columns, Nk = " << Nk << " (32-bit) words i.e. CipherKeyLenghth = "
+        << CipherKeyLenghth << " bytes (or " << CipherKeyLenghth * 8 << " bits), Nr = " << Nr << " rounds" << endl << endl;
+	*/
+
+    // create a dummy test cipher key
+    unsigned char key[CipherKeyLenghth_max];
+	cout << "key = ";
+    for (unsigned short i = 0; i < CipherKeyLenghth; i++) {
+        key[i] = (unsigned char)i;
+    //    cout << "0x" << hex << (unsigned short)key[i] << " ";
+    } //  cout << endl << endl;
+
+
+    // extend key
+    unsigned char expandedKey[ExtdCipherKeyLenghth_max];
+    KeyExpansion(key, Nk, expandedKey);
+	/*  cout << "expandedKey = ";
+    for (unsigned short i = 0; i < ExtdCipherKeyLenghth; i++) { cout << dec << (unsigned short)expandedKey[i] << " "; }
+    cout << "<=> ";
+    for (unsigned short i = 0; i < ExtdCipherKeyLenghth; i++) { cout << "0x" << hex << (unsigned short)expandedKey[i] << " "; }
+    cout << endl << endl;
+	*/
+
+
+	//Read Input Bitstream
+	streampos input_file_size;
+	char *data_in_file;
+	unsigned int plaintext[16];
+	int dec_array_size=0;
+	int dec_file_size;
+	int dec_data;
+	ifstream file ("/research/ece/lnis/USERS/sampathkumar/OpenFPGA/openfpga_flow/tasks/fpga_bitstream
+					/generate_bitstream/device_auto/run001/k6_frac_N10_tileable_40nm/and2/MIN_ROUTE_CHAN_WIDTH/fabric_bitstream.txt",ios::in|ios::ate);
+	if (file.is_open())
```

Comment:
Please use the file handler in openfpga util library:

https://github.com/lnis-uofu/OpenFPGA/blob/master/libopenfpga/libopenfpgautil/src/openfpga_digest.h